packages/util-waiter/src/poller.ts

```diff
@@ -0,0 +1,36 @@
+import { sleep } from "./utils/sleep";
+import { WaiterOptions, WaiterResult, WaiterState } from "./waiter";
+
+function exponentialBackoff(floor: number, ciel: number, attempt: number): number {
+  return Math.min(ciel, floor * 2 ** attempt);
+}
```

Comment:
```suggestion
const exponentialBackoff = (floor: number, ciel: number, attempt: number) => Math.min(ciel, floor * 2 ** attempt);
```


You can also import from [middleware-retry package](https://github.com/aws/aws-sdk-js-v3/blob/1f4b7cfbb801b139e52ca6814e344234599221d9/packages/middleware-retry/src/delayDecider.ts). It won't increase the bundle size as each client already includes this dependency. It does a little more than this too -- round the return value.

Reply:
The `defaultDelayDecider` in middleware-retry package uses fully-jittered exponential backoff which includes using `Math.random()` and can't be used here.

Question (can be followed up in a separate GitHub issue): Should waiters also use fully-jittered exponential backoff time like retry?

Reply:
Smithy has adopted jitters in waiters in https://github.com/awslabs/smithy/pull/656